niworkflows/workflows/epi/refmap.py

```diff
@@ -0,0 +1,148 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""Workflow for the generation of EPI (echo-planar imaging) references."""
+
+from nipype.pipeline import engine as pe
+from nipype.interfaces import utility as niu
+
+from ...engine.workflows import LiterateWorkflow as Workflow
+
+
+DEFAULT_MEMORY_MIN_GB = 0.01
+
+
+def init_epi_reference_wf(omp_nthreads, name="epi_reference_wf"):
+    """
+    Build a workflow that generates a reference map from a set of EPI images.
+
+    .. danger ::
+
+        All input files MUST have the same shimming configuration.
+        At the very least, make sure all input EPI images are acquired within the
+        same session, and have the same PE direction and total readout time.
+
+    Parameters
+    ----------
+    omp_nthreads : :obj:`int`
+        Maximum number of threads an individual process may use
+    name : :obj:`str`
+        Name of workflow (default: ``epi_reference_wf``)
+
+    Inputs
+    ------
+    in_files : :obj:`list` of :obj:`str`
+        List of paths of the input EPI images from which reference volumes will be
+        selected, aligned and averaged.
+
+    Outputs
+    -------
+    out_epiref : :obj:`str`
+        Path of the generated EPI reference file.
+
+    See Also
+    --------
+    Discussion and original flowchart at `nipreps/niworkflows#601
+    <https://github.com/nipreps/niworkflows/issues/601>`__.
+
+    """
+    from nipype.interfaces.ants import N4BiasFieldCorrection
+    from ...utils.connections import listify
+
+    from ...interfaces.bold import NonsteadyStatesDetector
+    from ...interfaces.freesurfer import StructuralReference
+    from ...interfaces.header import ValidateImage
+    from ...interfaces.images import RobustAverage
+    from ...interfaces.nibabel import IntensityClip
+
+    wf = Workflow(name=name)
+    inputnode = pe.Node(niu.IdentityInterface(fields=["in_files"]), name="inputnode")
+    outputnode = pe.Node(
+        niu.IdentityInterface(fields=["epiref", "xfms", "volumes"]), name="outputnode"
+    )
+
+    validate_nii = pe.MapNode(
+        ValidateImage(), name="validate_nii", iterfield=["in_file"]
+    )
+
+    # TODO: allow b=0 selection for DWI datasets instead
+    select_volumes = pe.MapNode(
+        NonsteadyStatesDetector(), name="select_volumes", iterfield=["in_file"]
+    )
+    run_avgs = pe.MapNode(
+        RobustAverage(), name="run_avgs", mem_gb=1, iterfield=["in_file", "t_mask"]
+    )
+
+    clip_avgs = pe.MapNode(IntensityClip(), name="clip_avgs", iterfield=["in_file"])
+
+    # de-gradient the fields ("bias/illumination artifact")
+    n4_avgs = pe.MapNode(
+        N4BiasFieldCorrection(
```

Comment:
These settings aren't working very well for rodents, will I put together a PR that allows an optional default override?

Reply:
yup, I had to start somewhere :)

Reply:
Of course, I was just checking you didn't already have something planned 👌🏼

Reply:
#611